new.cpp

```
@@ -0,0 +1,32 @@
+#include<iostream>
+using namespace std;
+int* productExceptSelf(int* nums, int numsSize) {
+    int *ret=(int*)malloc(numsSize*sizeof(int));
+    ret[0]=1;
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)